applications/tari_merge_mining_proxy/src/main.rs

```diff
@@ -226,6 +242,60 @@ fn get_request_id(json: &[u8]) -> i64 {
     parsed["id"].as_i64().unwrap_or_else(|| -1)
 }
 
+fn get_monero_difficulty(data: &[u8]) -> Option<u64> {
+    debug!(target: LOG_TARGET, "Data {}", &stringify_request(data));
+    let parsed = json::parse(&stringify_request(data));
+    match parsed {
+        Ok(parsed) => {
+            if parsed["result"]["difficulty"].is_null() {
+                return None;
+            }
+            let difficulty = &parsed["result"]["difficulty"].as_u64();
+            difficulty.to_owned()
+        },
+        Err(e) => {
+            error!(
+                target: LOG_TARGET,
+                "{:#}",
+                MmProxyError::ParseError(format!("Failure to parse json to get monero difficulty, {:?}", e))
+            );
+            None
+        },
+    }
+}
+
+fn select_and_update_lowest_difficulty(data: &[u8], tari_difficulty: u64) -> Vec<u8> {
+    let monero_difficulty = get_monero_difficulty(data);
+    match monero_difficulty {
+        Some(monero_difficulty) => {
+            let parsed = json::parse(&stringify_request(data));
+            match parsed {
+                Ok(mut parsed) => {
+                    let difficulty = min(monero_difficulty, tari_difficulty).to_owned();
```

Comment:
This will favour the min. 
You are comparing Option<u64> here. 
None is less than some(1). 
```suggestion
                    let difficulty = min(monero_difficulty.unwrap_or(u64::max), tari_difficulty.unwrap_or(u64::max));
```

Reply:
Ideally if both at broken, we cant mine so we should return error.

Reply:
Neither are options here, tari_difficulty is passed in as u64,  monero is matched before it reaches this line.

Reply:
O sorry, youre right, I missed the match above.